File: apps/api/app/api/v1/workbench.py
from __future__ import annotations

from pathlib import Path
from typing import Any

from fastapi import APIRouter, HTTPException

from apps.api.app.application.workbench_service import WorkbenchService
from apps.api.app.domain.workbench.entities import (
    AgentRun,
    Asset,
    LiveComponent,
    LiveRoomComposition,
    LiveScene,
    PlatformMetricSnapshot,
    Project,
    WorkflowRule,
    WorkflowRun,
)

router = APIRouter(tags=["workbench"])
_service = WorkbenchService(Path(__file__).resolve().parents[5])


@router.get("/dashboard/summary")
def dashboard_summary() -> dict[str, Any]:
    return _service.dashboard_summary()


@router.get("/products")
def list_products() -> dict[str, Any]:
    return {"products": [item.model_dump() for item in _service.products.list()]}


@router.post("/products")
def create_product(payload: dict[str, Any]) -> dict[str, Any]:
    return {"product": _service.create_product(payload).model_dump()}


@router.get("/products/{product_id}")
def get_product(product_id: str) -> dict[str, Any]:
    try:
        return {"product": _service.products.get(product_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.patch("/products/{product_id}")
def update_product(product_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"product": _service.update_product(product_id, payload).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.delete("/products/{product_id}")
def delete_product(product_id: str) -> dict[str, str]:
    _service.products.delete(product_id)
    return {"status": "deleted"}


@router.post("/products/{product_id}/publish")
def publish_product(product_id: str) -> dict[str, Any]:
    try:
        return {"product": _service.publish_product(product_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.post("/products/{product_id}/unpublish")
def unpublish_product(product_id: str) -> dict[str, Any]:
    try:
        return {"product": _service.unpublish_product(product_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/avatars")
def list_avatars() -> dict[str, Any]:
    return {"avatars": [item.model_dump() for item in _service.avatars.list()]}


@router.post("/avatars")
def create_avatar(payload: dict[str, Any]) -> dict[str, Any]:
    return {"avatar": _service.create_avatar(payload).model_dump()}


@router.patch("/avatars/{avatar_id}")
def update_avatar(avatar_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"avatar": _service.update_avatar(avatar_id, payload).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/scripts/templates")
def list_script_templates() -> dict[str, Any]:
    return {"templates": [item.model_dump() for item in _service.scripts.list()]}


@router.post("/scripts/templates")
def create_script_template(payload: dict[str, Any]) -> dict[str, Any]:
    return {"template": _service.create_script(payload).model_dump()}


@router.patch("/scripts/templates/{template_id}")
def update_script_template(template_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"template": _service.update_script(template_id, payload).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.delete("/scripts/templates/{template_id}")
def delete_script_template(template_id: str) -> dict[str, str]:
    _service.scripts.delete(template_id)
    return {"status": "deleted"}


@router.post("/scripts/templates/generate")
def generate_script_template(payload: dict[str, Any]) -> dict[str, Any]:
    category = str(payload.get("category") or "interaction")
    product_id = str(payload.get("product_id") or "")
    return {"template": _service.generate_script(category, product_id).model_dump()}


@router.get("/workflow/rules")
def list_workflow_rules() -> dict[str, Any]:
    return {"rules": [item.model_dump() for item in _service.workflow_rules.list()]}


@router.post("/workflow/rules")
def create_workflow_rule(payload: dict[str, Any]) -> dict[str, Any]:
    return {"rule": _service.workflow_rules.upsert(WorkflowRule.model_validate(payload)).model_dump()}


@router.patch("/workflow/rules/{rule_id}")
def update_workflow_rule(rule_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        rule = _service.workflow_rules.get(rule_id)
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc
    return {"rule": _service.workflow_rules.upsert(rule.model_copy(update=payload)).model_dump()}


@router.delete("/workflow/rules/{rule_id}")
def delete_workflow_rule(rule_id: str) -> dict[str, str]:
    _service.workflow_rules.delete(rule_id)
    return {"status": "deleted"}


@router.get("/platform/accounts")
def list_platform_accounts() -> dict[str, Any]:
    return {"accounts": [item.model_dump() for item in _service.platform_accounts.list()]}


@router.get("/platform/metrics")
def list_platform_metrics() -> dict[str, Any]:
    return {"metrics": [item.model_dump() for item in _service.metrics.list()]}


@router.post("/platform/metrics")
def create_platform_metric(payload: dict[str, Any]) -> dict[str, Any]:
    return {"metric": _service.metrics.upsert(PlatformMetricSnapshot.model_validate(payload)).model_dump()}


@router.get("/knowledge/documents")
def list_knowledge_documents() -> dict[str, Any]:
    return {"documents": [item.model_dump() for item in _service.knowledge_documents.list()]}


@router.post("/knowledge/documents")
def create_knowledge_document(payload: dict[str, Any]) -> dict[str, Any]:
    return {"document": _service.create_knowledge_document(payload).model_dump()}


@router.delete("/knowledge/documents/{document_id}")
def delete_knowledge_document(document_id: str) -> dict[str, str]:
    _service.knowledge_documents.delete(document_id)
    remaining_chunks = [chunk for chunk in _service.knowledge_chunks.list() if chunk.document_id != document_id]
    _service.knowledge_chunks._write(remaining_chunks)
    return {"status": "deleted"}


@router.post("/knowledge/documents/{document_id}/index")
def index_knowledge_document(document_id: str, payload: dict[str, Any] | None = None) -> dict[str, Any]:
    try:
        document = _service.index_knowledge_document(document_id, str((payload or {}).get("text") or ""))
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc
    return {"document": document.model_dump(), "chunks": [item.model_dump() for item in _service.knowledge_chunks.list() if item.document_id == document_id]}


@router.post("/knowledge/search")
def search_knowledge(payload: dict[str, Any]) -> dict[str, Any]:
    results = _service.search_knowledge_with_scores(
        str(payload.get("query") or ""),
        product_id=str(payload.get("product_id") or ""),
        limit=int(payload.get("limit") or 5),
    )
    return {
        "chunks": [item["chunk"].model_dump() for item in results],
        "results": [{"score": item["score"], "matched_terms": item["matched_terms"], "chunk": item["chunk"].model_dump()} for item in results],
    }


@router.get("/model-gateway/providers")
def list_model_providers() -> dict[str, Any]:
    return {"providers": [item.model_dump() for item in _service.model_providers.list()]}


@router.get("/model-gateway/prompts")
def list_prompt_templates() -> dict[str, Any]:
    return {"prompts": [item.model_dump() for item in _service.prompt_templates.list()]}


@router.post("/avatars/{avatar_id}/jobs")
def create_avatar_job(avatar_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        _service.avatars.get(avatar_id)
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc
    return {"job": _service.create_avatar_job({**payload, "avatar_id": avatar_id}).model_dump()}


@router.get("/avatars/jobs/{job_id}")
def get_avatar_job(job_id: str) -> dict[str, Any]:
    try:
        return {"job": _service.avatar_jobs.get(job_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/platform/events")
def list_platform_events() -> dict[str, Any]:
    return {"events": [item.model_dump() for item in _service.platform_events.list()]}


@router.post("/platform/events")
def ingest_platform_event(payload: dict[str, Any]) -> dict[str, Any]:
    return {"event": _service.ingest_platform_event(payload).model_dump()}


@router.get("/projects")
def list_projects() -> dict[str, Any]:
    return {"projects": [item.model_dump() for item in _service.projects.list()]}


@router.post("/projects")
def create_project(payload: dict[str, Any]) -> dict[str, Any]:
    return {"project": _service.create_project(payload).model_dump()}


@router.get("/mvp/live-plans")
def list_mvp_live_plans(project_id: str = "") -> dict[str, Any]:
    return {"plans": [item.model_dump() for item in _service.list_mvp_live_plans(project_id)]}


@router.get("/mvp/live-plans/{plan_id}")
def get_mvp_live_plan(plan_id: str) -> dict[str, Any]:
    try:
        return {"plan": _service.get_mvp_live_plan(plan_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.post("/mvp/live-plans/run")
def run_mvp_live_commerce(payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"plan": _service.run_mvp_live_commerce(payload).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/projects/{project_id}")
def get_project(project_id: str) -> dict[str, Any]:
    try:
        return {"project": _service.projects.get(project_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.patch("/projects/{project_id}")
def update_project(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"project": _service.update_project(project_id, payload).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/agents")
def list_agents() -> dict[str, Any]:
    return {"agents": [item.model_dump() for item in _service.agent_profiles.list()]}


@router.get("/agents/runs")
def list_all_agent_runs() -> dict[str, Any]:
    return {"runs": [item.model_dump() for item in _service.agent_runs.list()]}


@router.get("/assets")
def list_all_assets() -> dict[str, Any]:
    return {"assets": [item.model_dump() for item in _service.assets.list()]}


@router.get("/components")
def list_all_components() -> dict[str, Any]:
    return {"components": [item.model_dump() for item in _service.components.list()]}


@router.get("/scenes")
def list_all_live_scenes() -> dict[str, Any]:
    return {"scenes": [item.model_dump() for item in _service.live_scenes.list()]}


@router.get("/live-room-compositions")
def list_all_live_room_compositions() -> dict[str, Any]:
    return {"compositions": [item.model_dump() for item in _service.live_room_compositions.list()]}


@router.get("/workflow/definitions")
def list_workflow_definitions() -> dict[str, Any]:
    return {"definitions": [item.model_dump() for item in _service.workflow_definitions.list()]}


@router.get("/workflow/runs")
def list_all_workflow_runs() -> dict[str, Any]:
    return {"runs": [item.model_dump() for item in _service.workflow_runs.list()]}


@router.get("/workflow/runs/{workflow_run_id}/nodes")
def list_workflow_node_runs(workflow_run_id: str) -> dict[str, Any]:
    return {"nodes": [item.model_dump() for item in _service.workflow_node_runs.list() if item.workflow_run_id == workflow_run_id]}


@router.get("/prompt-versions")
def list_prompt_versions() -> dict[str, Any]:
    return {"prompt_versions": [item.model_dump() for item in _service.prompt_versions.list()]}


@router.get("/plugins/providers")
def list_plugin_providers(category: str = "") -> dict[str, Any]:
    return {"providers": [item.model_dump() for item in _service.list_plugins(category)]}


@router.get("/plugins/providers/{provider_id}/health")
def get_plugin_health(provider_id: str) -> dict[str, Any]:
    try:
        return {"health": _service.plugin_health(provider_id)}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.post("/plugins/providers/{provider_id}/estimate")
def estimate_plugin_cost(provider_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"estimate": _service.estimate_plugin_cost(provider_id, payload)}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.post("/plugins/providers/{provider_id}/jobs")
def submit_plugin_job(provider_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    try:
        return {"job": _service.submit_plugin_job(provider_id, payload)}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/plugins/providers/{provider_id}/jobs/{job_id}")
def get_plugin_job(provider_id: str, job_id: str) -> dict[str, Any]:
    try:
        return {"job": _service.get_plugin_job(provider_id, job_id)}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.post("/plugins/providers/{provider_id}/jobs/{job_id}/cancel")
def cancel_plugin_job(provider_id: str, job_id: str) -> dict[str, Any]:
    try:
        return {"job": _service.cancel_plugin_job(provider_id, job_id)}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/analytics/overview")
def analytics_overview() -> dict[str, Any]:
    return _service.analytics_overview()


@router.get("/projects/{project_id}/agent-runs")
def list_project_agent_runs(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"runs": [item.model_dump() for item in _service.list_agent_runs(project_id)]}


@router.post("/projects/{project_id}/agent-runs")
def create_project_agent_run(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"run": _service.agent_runs.upsert(AgentRun.model_validate({**payload, "project_id": project_id})).model_dump()}


@router.get("/projects/{project_id}/assets")
def list_project_assets(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"assets": [item.model_dump() for item in _service.list_assets(project_id)]}


@router.post("/projects/{project_id}/assets")
def create_project_asset(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"asset": _service.create_asset(project_id, payload).model_dump()}


@router.get("/projects/{project_id}/components")
def list_project_components(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"components": [item.model_dump() for item in _service.list_components(project_id)]}


@router.post("/projects/{project_id}/components")
def create_project_component(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"component": _service.create_component(project_id, payload).model_dump()}


@router.get("/projects/{project_id}/scenes")
def list_project_scenes(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"scenes": [item.model_dump() for item in _service.list_live_scenes(project_id)]}


@router.post("/projects/{project_id}/scenes")
def create_project_scene(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"scene": _service.create_live_scene(project_id, payload).model_dump()}


@router.get("/projects/{project_id}/live-rooms")
def list_project_live_rooms(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"compositions": [item.model_dump() for item in _service.list_live_room_compositions(project_id)]}


@router.post("/projects/{project_id}/live-rooms")
def create_project_live_room(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"composition": _service.create_live_room(project_id, payload).model_dump()}


@router.get("/projects/{project_id}/workflow-runs")
def list_project_workflow_runs(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"runs": [item.model_dump() for item in _service.list_workflow_runs(project_id)]}


@router.post("/projects/{project_id}/workflow-runs")
def create_project_workflow_run(project_id: str, payload: dict[str, Any]) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"run": _service.workflow_runs.upsert(WorkflowRun.model_validate({**payload, "project_id": project_id})).model_dump()}


@router.get("/projects/{project_id}/analytics/best-practices")
def list_project_best_practices(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return {"best_practices": [item.model_dump() for item in _service.list_best_practices(project_id)]}


@router.post("/projects/{project_id}/analytics/best-practices/{best_practice_id}/clone")
def clone_project_best_practice(project_id: str, best_practice_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    try:
        return {"composition": _service.clone_best_practice(best_practice_id, target_project_id=project_id).model_dump()}
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/projects/{project_id}/analytics/overview")
def project_analytics_overview(project_id: str) -> dict[str, Any]:
    _ensure_project(project_id)
    return _service.analytics_overview(project_id)


def _ensure_project(project_id: str) -> Project:
    try:
        return _service.projects.get(project_id)
    except KeyError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc

File: apps/web/lib/api/workbench.ts
import { apiRequest } from '@/lib/api/config';
import type { AgentProfile, AgentRun, AnalyticsOverview, Asset, AvatarProfile, BestPractice, DashboardSummary, KnowledgeChunk, KnowledgeDocument, LiveComponent, LiveRoomComposition, LiveScene, ModelProviderConfig, MvpLivePlan, PlatformEvent, PlatformMetricSnapshot, PluginProvider, ProductRecord, Project, PromptTemplate, PromptVersion, ScriptTemplate, WorkflowDefinition, WorkflowNodeRun, WorkflowRule, WorkflowRun } from '@/types/workbench';

export async function getDashboardSummary(): Promise<DashboardSummary> {
  return apiRequest<DashboardSummary>('/api/v1/dashboard/summary');
}

export async function listProducts(): Promise<ProductRecord[]> {
  const payload = await apiRequest<{ products: ProductRecord[] }>('/api/v1/products');
  return payload.products;
}

export async function createProduct(payload: Partial<ProductRecord>): Promise<ProductRecord> {
  const response = await apiRequest<{ product: ProductRecord }>('/api/v1/products', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.product;
}

export async function publishProduct(productId: string): Promise<ProductRecord> {
  const response = await apiRequest<{ product: ProductRecord }>(`/api/v1/products/${productId}/publish`, { method: 'POST' });
  return response.product;
}

export async function unpublishProduct(productId: string): Promise<ProductRecord> {
  const response = await apiRequest<{ product: ProductRecord }>(`/api/v1/products/${productId}/unpublish`, { method: 'POST' });
  return response.product;
}

export async function listAvatars(): Promise<AvatarProfile[]> {
  const payload = await apiRequest<{ avatars: AvatarProfile[] }>('/api/v1/avatars');
  return payload.avatars;
}

export async function createAvatar(payload: Partial<AvatarProfile>): Promise<AvatarProfile> {
  const response = await apiRequest<{ avatar: AvatarProfile }>('/api/v1/avatars', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.avatar;
}

export async function createAvatarJob(avatarId: string, inputText: string): Promise<{ job_id: string; status: string; output_url: string }> {
  const response = await apiRequest<{ job: { job_id: string; status: string; output_url: string } }>(`/api/v1/avatars/${avatarId}/jobs`, {
    method: 'POST',
    body: JSON.stringify({ input_text: inputText, job_type: 'text_drive' }),
  });
  return response.job;
}

export async function listScriptTemplates(): Promise<ScriptTemplate[]> {
  const payload = await apiRequest<{ templates: ScriptTemplate[] }>('/api/v1/scripts/templates');
  return payload.templates;
}

export async function generateScriptTemplate(category: string, productId = ''): Promise<ScriptTemplate> {
  const payload = await apiRequest<{ template: ScriptTemplate }>('/api/v1/scripts/templates/generate', {
    method: 'POST',
    body: JSON.stringify({ category, product_id: productId }),
  });
  return payload.template;
}

export async function listWorkflowRules(): Promise<WorkflowRule[]> {
  const payload = await apiRequest<{ rules: WorkflowRule[] }>('/api/v1/workflow/rules');
  return payload.rules;
}

export async function createWorkflowRule(payload: Partial<WorkflowRule>): Promise<WorkflowRule> {
  const response = await apiRequest<{ rule: WorkflowRule }>('/api/v1/workflow/rules', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.rule;
}

export async function updateWorkflowRule(ruleId: string, payload: Partial<WorkflowRule>): Promise<WorkflowRule> {
  const response = await apiRequest<{ rule: WorkflowRule }>(`/api/v1/workflow/rules/${ruleId}`, {
    method: 'PATCH',
    body: JSON.stringify(payload),
  });
  return response.rule;
}

export async function listPlatformMetrics(): Promise<PlatformMetricSnapshot[]> {
  const payload = await apiRequest<{ metrics: PlatformMetricSnapshot[] }>('/api/v1/platform/metrics');
  return payload.metrics;
}

export async function listKnowledgeDocuments(): Promise<KnowledgeDocument[]> {
  const payload = await apiRequest<{ documents: KnowledgeDocument[] }>('/api/v1/knowledge/documents');
  return payload.documents;
}

export async function createKnowledgeDocument(payload: Partial<KnowledgeDocument> & { text?: string }): Promise<KnowledgeDocument> {
  const response = await apiRequest<{ document: KnowledgeDocument }>('/api/v1/knowledge/documents', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  if (payload.text) {
    const indexed = await apiRequest<{ document: KnowledgeDocument }>(`/api/v1/knowledge/documents/${response.document.document_id}/index`, {
      method: 'POST',
      body: JSON.stringify({ text: payload.text }),
    });
    return indexed.document;
  }
  return response.document;
}

export async function searchKnowledge(query: string): Promise<KnowledgeChunk[]> {
  const payload = await apiRequest<{ chunks: KnowledgeChunk[] }>('/api/v1/knowledge/search', {
    method: 'POST',
    body: JSON.stringify({ query }),
  });
  return payload.chunks;
}

export async function listModelProviders(): Promise<ModelProviderConfig[]> {
  const payload = await apiRequest<{ providers: ModelProviderConfig[] }>('/api/v1/model-gateway/providers');
  return payload.providers;
}

export async function listPromptTemplates(): Promise<PromptTemplate[]> {
  const payload = await apiRequest<{ prompts: PromptTemplate[] }>('/api/v1/model-gateway/prompts');
  return payload.prompts;
}

export async function createPromptTemplate(payload: { name: string; system: string; user_instruction: string; max_output_seconds?: number | null }): Promise<PromptTemplate> {
  const response = await apiRequest<{ prompt: PromptTemplate }>('/api/v1/model-gateway/prompts', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.prompt;
}

export async function renderPromptTemplate(promptTemplate: string, promptPayload: Record<string, unknown>): Promise<{ role: string; content: string }[]> {
  const response = await apiRequest<{ messages: { role: string; content: string }[] }>('/api/v1/model-gateway/prompts/render', {
    method: 'POST',
    body: JSON.stringify({ prompt_template: promptTemplate, prompt_payload: promptPayload }),
  });
  return response.messages;
}

export async function listPlatformEvents(): Promise<PlatformEvent[]> {
  const payload = await apiRequest<{ events: PlatformEvent[] }>('/api/v1/platform/events');
  return payload.events;
}

export async function createPlatformEvent(payload: Partial<PlatformEvent>): Promise<PlatformEvent> {
  const response = await apiRequest<{ event: PlatformEvent }>('/api/v1/platform/events', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.event;
}

export async function createPlatformMetric(payload: Partial<PlatformMetricSnapshot>): Promise<PlatformMetricSnapshot> {
  const response = await apiRequest<{ metric: PlatformMetricSnapshot }>('/api/v1/platform/metrics', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.metric;
}

export async function listProjects(): Promise<Project[]> {
  const payload = await apiRequest<{ projects: Project[] }>('/api/v1/projects');
  return payload.projects;
}

export async function listMvpLivePlans(projectId = ''): Promise<MvpLivePlan[]> {
  const query = projectId ? `?project_id=${encodeURIComponent(projectId)}` : '';
  const payload = await apiRequest<{ plans: MvpLivePlan[] }>(`/api/v1/mvp/live-plans${query}`);
  return payload.plans;
}

export async function runMvpLivePlan(payload: { project_id?: string; product_id?: string; product?: Record<string, unknown>; brand_name?: string; script_note?: string; avatar_id?: string }): Promise<MvpLivePlan> {
  const response = await apiRequest<{ plan: MvpLivePlan }>('/api/v1/mvp/live-plans/run', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.plan;
}

export async function createProject(payload: Partial<Project>): Promise<Project> {
  const response = await apiRequest<{ project: Project }>('/api/v1/projects', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.project;
}

export async function listAgents(): Promise<AgentProfile[]> {
  const payload = await apiRequest<{ agents: AgentProfile[] }>('/api/v1/agents');
  return payload.agents;
}

export async function listAgentRuns(projectId?: string): Promise<AgentRun[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/agent-runs` : '/api/v1/agents/runs';
  const payload = await apiRequest<{ runs: AgentRun[] }>(path);
  return payload.runs;
}

export async function listAssets(projectId?: string): Promise<Asset[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/assets` : '/api/v1/assets';
  const payload = await apiRequest<{ assets: Asset[] }>(path);
  return payload.assets;
}

export async function createAsset(projectId: string, payload: Partial<Asset>): Promise<Asset> {
  const response = await apiRequest<{ asset: Asset }>(`/api/v1/projects/${projectId}/assets`, {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.asset;
}

export async function listComponents(projectId?: string): Promise<LiveComponent[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/components` : '/api/v1/components';
  const payload = await apiRequest<{ components: LiveComponent[] }>(path);
  return payload.components;
}

export async function createComponent(projectId: string, payload: Partial<LiveComponent>): Promise<LiveComponent> {
  const response = await apiRequest<{ component: LiveComponent }>(`/api/v1/projects/${projectId}/components`, {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response.component;
}

export async function listLiveScenes(projectId?: string): Promise<LiveScene[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/scenes` : '/api/v1/scenes';
  const payload = await apiRequest<{ scenes: LiveScene[] }>(path);
  return payload.scenes;
}

export async function listLiveRoomCompositions(projectId?: string): Promise<LiveRoomComposition[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/live-rooms` : '/api/v1/live-room-compositions';
  const payload = await apiRequest<{ compositions: LiveRoomComposition[] }>(path);
  return payload.compositions;
}

export async function listWorkflowDefinitions(): Promise<WorkflowDefinition[]> {
  const payload = await apiRequest<{ definitions: WorkflowDefinition[] }>('/api/v1/workflow/definitions');
  return payload.definitions;
}

export async function listWorkflowRuns(projectId?: string): Promise<WorkflowRun[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/workflow-runs` : '/api/v1/workflow/runs';
  const payload = await apiRequest<{ runs: WorkflowRun[] }>(path);
  return payload.runs;
}

export async function listWorkflowNodeRuns(workflowRunId: string): Promise<WorkflowNodeRun[]> {
  const payload = await apiRequest<{ nodes: WorkflowNodeRun[] }>(`/api/v1/workflow/runs/${workflowRunId}/nodes`);
  return payload.nodes;
}

export async function listBestPractices(projectId?: string): Promise<BestPractice[]> {
  const path = projectId ? `/api/v1/projects/${projectId}/analytics/best-practices` : '/api/v1/analytics/overview';
  if (projectId) {
    const payload = await apiRequest<{ best_practices: BestPractice[] }>(path);
    return payload.best_practices;
  }
  const payload = await apiRequest<AnalyticsOverview>(path);
  return payload.best_practices;
}

export async function getAnalyticsOverview(projectId?: string): Promise<AnalyticsOverview> {
  const path = projectId ? `/api/v1/projects/${projectId}/analytics/overview` : '/api/v1/analytics/overview';
  return apiRequest<AnalyticsOverview>(path);
}

export async function listPromptVersions(): Promise<PromptVersion[]> {
  const payload = await apiRequest<{ prompt_versions: PromptVersion[] }>('/api/v1/prompt-versions');
  return payload.prompt_versions;
}

export async function listPluginProviders(): Promise<PluginProvider[]> {
  const payload = await apiRequest<{ providers: PluginProvider[] }>('/api/v1/plugins/providers');
  return payload.providers;
}
